File: leetcode/editor/en/[70]Climbing Stairs.java
70
        Climbing Stairs
        2022-12-14 15:49:30

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    Map<Integer, Integer> map = new HashMap<>();

    public int climbStairs(int n) {
        return dp(n);
    }

    private int dp(int i) {
        if (i <= 2) {
            return i;
        }
        if (!map.containsKey(i)) {
            map.put(i, dp(i - 1) + dp(i - 2));
        }
        return map.get(i);
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[70]Climbing Stairs.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n) {
    let dp = [1, 1];
    for (let i = 2; i < n + 1; i++) {
        dp[i] = dp[i - 1] + dp[i - 2];
    }
    return dp[n];
};
//leetcode submit region end(Prohibit modification and deletion)
